apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/utility/DataUtilityCommons.java

```diff
@@ -0,0 +1,311 @@
+    }
+
+    Path originalNpiDataFile = workingDir.resolve(matchingFiles[0].getName());
+    if (!Files.isReadable(originalNpiDataFile))
+      throw new IllegalStateException("Unable to locate npidata_pfile in " + ndctextZipUrl);
+    return originalNpiDataFile;
+  }
+
+  /**
+   * Deletes the directory.
+   *
+   * @param tempDir for the temp directory
+   */
+  private static void recursivelyDelete(Path tempDir) {
+    // Recursively delete the working dir.
+    try {
+      Files.walk(tempDir)
+          .sorted(Comparator.reverseOrder())
+          .map(Path::toFile)
+          .peek(System.out::println)
+          .forEach(File::delete);
+    } catch (IOException e) {
+      LOGGER.warn("Failed to cleanup the temporary folder", e);
+    }
+  }
+
+  /**
+   * Creates the {@link #NPI_RESOURCE} file in the specified location.
+   *
+   * @param convertedNpiDataFile the output file/resource to produce
+   * @param workingDir a directory that temporary/working files can be written to
+   * @throws IOException (any errors encountered will be bubbled up)
+   */
+  private static void buildNPIResource(Path convertedNpiDataFile, Path workingDir)
+      throws IOException {
+    Path originalNpiDataFile;
+    String fileName;
+
+    try {
+      fileName = getFileName(false);
+      originalNpiDataFile = getOriginalNpiDataFile(workingDir, fileName);
+    } catch (IOException e) {
+      fileName = getFileName(true);
+      originalNpiDataFile = getOriginalNpiDataFile(workingDir, fileName);
+    }
+    convertNpiDataFile(convertedNpiDataFile, originalNpiDataFile);
+  }
+
+  /**
+   * Converts the npi data file.
+   *
+   * @param convertedNpiDataFile converted npi data file.
+   * @param originalNpiDataFile original npi data file.
+   * @throws IOException exception thrown.
+   */
+  private static void convertNpiDataFile(Path convertedNpiDataFile, Path originalNpiDataFile)
+      throws IOException {
+    // convert file format from cp1252 to utf8
+    CharsetDecoder inDec =
+        Charset.forName("windows-1252")
+            .newDecoder()
+            .onMalformedInput(CodingErrorAction.REPORT)
+            .onUnmappableCharacter(CodingErrorAction.REPORT);
+
+    CharsetEncoder outEnc =
+        StandardCharsets.UTF_8
+            .newEncoder()
+            .onMalformedInput(CodingErrorAction.REPORT)
+            .onUnmappableCharacter(CodingErrorAction.REPORT);
+
+    try (FileInputStream is = new FileInputStream(originalNpiDataFile.toFile().getAbsolutePath());
+        BufferedReader reader = new BufferedReader(new InputStreamReader(is, inDec));
+        FileOutputStream fw =
+            new FileOutputStream(convertedNpiDataFile.toFile().getAbsolutePath());
+        BufferedWriter out = new BufferedWriter(new OutputStreamWriter(fw, outEnc))) {
+
+      // skip first line which is header
+      reader.readLine();
+      for (String line; (line = reader.readLine()) != null; ) {
+        String[] fields = line.split(",");
```

Comment:
should you at least check the count of entries in the fields[] array since you then grab elements of the array by index number.